### PR TITLE
feat: Update Announcement Screen design and add new fields

### LINKED ISSUE #230: add category field and availability list in annoucement screen

### PR DESCRIPTION
**Pull request by daferya**
- Redesigned the Announcement screen for improved user experience.
- Added an availability list field to allow users to specify time slots.
- Introduced a category field for better classification of announcements.
- Updated tests in `AnnouncementUserNoModeScreenTest.kt` to align with the new design and functionality.
- Adjusted `QuickFixFinder` to support the updated Announcement screen design and logic.

Related to issue #230.

### CODE REVIEW

**Review by AnotherMedo — CHANGES_REQUESTED** [review]
Great addition! there are a lot of absolute paddings here and there to refactor however.

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/search/Announcement.kt [hunk 1]
```diff
@@ -143,368 +189,676 @@ fun AnnouncementScreen(
...
-    // Use Scaffold for the layout structure
+    val categoryTextStyle =
+        MaterialTheme.typography.labelMedium.copy(
+            fontSize = 10.sp, color = colorScheme.onBackground, fontWeight = FontWeight.Medium)
+
+    val maxCategoryTextWidth =
+        calculateMaxTextWidth(
+            texts = allSubcategories.map { it.name }, textStyle = categoryTextStyle)
+
+    val dropdownMenuWidth = maxCategoryTextWidth + 40.dp
```
[the remark this answers is not in the document]
Comment (daferya) [reply app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/search/Announcement.kt:266]: I apply when i use dropdownMenuWidth

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/search/Announcement.kt [hunk 2]
```diff
@@ -143,368 +189,676 @@ fun AnnouncementScreen(
...
           Column(
               modifier =
-                  Modifier.fillMaxWidth()
+                  Modifier.fillMaxSize()
                       .padding(padding)
-                      .padding(top = 30.dp)
-                      .align(Alignment.Center)
-                      .padding(16.dp)
-                      .zIndex(100f)
+                      .padding(start = 14.dp, end = 14.dp, top = 30.dp)
```
[the remark this answers is not in the document]
Comment (daferya) [reply app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/search/Announcement.kt:276]: Thank you for the feedback! I will make the necessary changes to replace the absolute values with constants or appropriate calculations to improve the maintainability of the code.

File: app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/search/QuickFixFinder.kt [hunk 3]
```diff
@@ -64,19 +69,18 @@ fun QuickFixFinderScreen(
...
             verticalArrangement = Arrangement.Center,
             horizontalAlignment = Alignment.CenterHorizontally) {
-              val pagerState = rememberPagerState(pageCount = { 2 })
               val coroutineScope = rememberCoroutineScope()
 
               Surface(
-                  color = colorScheme.surface,
+                  color = colorButton,
                   shape = RoundedCornerShape(20.dp),
                   modifier = Modifier.padding(horizontal = 40.dp).clip(RoundedCornerShape(20.dp))) {
```
[the remark this answers is not in the document]
Comment (daferya) [reply app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/search/QuickFixFinder.kt:85]: Thanks for the feedback! 😄 Actually, this code has already been updated in another branch that hasn't been merged yet. Once that branch is merged, this issue should be resolved

**Review by AnotherMedo — APPROVED** [review]
Everything looks good to me! Thank you for being reactive on the changes 😄

**Pull request merged**